connector/spanmetricsconnector/connector_test.go

```diff
@@ -1998,13 +1999,18 @@ func TestConnectorWithCardinalityLimit(t *testing.T) {
 
 	assert.NoError(t, connector.ConsumeTraces(context.Background(), traces))
 
-	metrics := connector.buildMetrics()
+	// first buildMetrics emit zero datapoint value
+	_ = connector.buildMetrics()
+
+	// mock send two batch of spans to this connector
```

Comment:
Why two batch of spans?

In my PR I just called `connector.buildMetrics()` twice (one to initialize to 0 and another one to get the values of the spans the connector consumed).

I don't have a strong opinion about this but it took me a while to get why the values for the datapoints where those... until I noticed about this line.

Reply:
I think using a single batch of spans would also work. The reason I used two batches of spans was to avoid falling into potential edge-case traps in the code.

I can provide more comments on that to make it more noticeable.

Reply:
Great! Thanks